### DISCUSSION TITLE
Any experiences using dm-writeboost with a BTRFS formatted disk?

### DISCUSSION THREAD

**Discussion by hradec** [akiradeveloper/dm-writeboost#229] [Q&A]
Hi everyone! 

I want to use dm-writeboost with some of my BTRFS disks, but I don't have spare disks to backup the data. 

I was wondering if anyone here have tried to add a dm-writeboost cache to a BTRFS disk before, and how was the experience? Did everything worked as expected? Is it still working today? 

I used enhanceIO cache for a few years with these BTRFS disks, and was a dream! Unfortunately enhanceIO is not working in newer kernel versions anymore, and it seem to be dead due to a big change in the IO subsystem in kernel 5.10 and up. 

So I'm considering switching to dm-writeboost, but since I don't have spare disks to backup my data, I would love to hear any experiences regarding dm-writeboost and BTRFS disks. 

btw, I don't use any btrfs raid setup. Just single disks formatted with BTRFS, using metada duplication on the same disk. 

cheers...
-H

**Comment by onlyjob** [comment]
I've used dm-writeboost with Btrfs and everything worked as expected. I had no serious issues with dm-writeboost and in my experience it was more reliable than EnhanceIO.

I don't recommend to begin using dm-writeboost (or any other SSD caching) with your only disk. Ideally you should have a disposable system (at least one HDD and a partition on SSD) to familiarise yourself with setup. Don't learn on your only precious file system - your success depends on your skills/experience and there are risks of data loss if you do anything wrong.

Although if you have no backup then your data is already disposable but then why do you seek reassurance here?
Invest in backup disk and make sure your data is safe before embarking on experimentation that can lead to data loss - that would be the safest option.
